plexapi/library.py

```diff
@@ -294,8 +294,17 @@ def add(self, name='', type='', agent='', scanner='', location='', language='en'
                   40:South Africa, 41:Spain, 42:Sweden, 43:Switzerland, 44:Taiwan, 45:Trinidad,
                   46:United Kingdom, 47:United States, 48:Uruguay, 49:Venezuela.
         """
-        part = '/library/sections?name=%s&type=%s&agent=%s&scanner=%s&language=%s&location=%s' % (
-            quote_plus(name), type, agent, quote_plus(scanner), language, quote_plus(location))  # noqa E126
+        if isinstance(location, str):
+            if not self._server.isBrowsable(location):
+                raise BadRequest('Path: %s does not exist.' % location)
+            location = [location]
+        if isinstance(location, list):
+            for path in location:
+                if not self._server.isBrowsable(path):
+                    raise BadRequest('Path: %s does not exist.' % path)
+        locations = [('location', l) for l in location]
```

Comment:
Create new locations list when checking browsable so you only loop once.

```suggestion
         _locations = []
        for path in location:
            if not self._server.isBrowsable(path):
                raise BadRequest('Path: %s does not exist.' % path)
            _locations.append(('location', path))
                
        part = '/library/sections?name=%s&type=%s&agent=%s&scanner=%s&language=%s&%s' % (
            quote_plus(name), type, agent, quote_plus(scanner), language, urlencode(_locations, doseq=True))
```